mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/binarystorage/GridFSDelegator.java

```diff
@@ -0,0 +1,103 @@
+ * @author Sergey Chernolyas &amp;sergey_chernolyas@gmail.com&amp;
+ */
+public class GridFSDelegator implements BinaryStorageDelegator {
+
+	/**
+	 * store binary content of field to GridFS
+	 *
+	 * @param mongoDatabase current instance of MongoDB
+	 * @param optionsContext options context
+	 * @param currentDocument current BSON document
+	 * @param fieldName fieldName with binary content
+	 * @param tuple tuple for processing
+	 */
+
+	@Override
+	public void storeContentToBinaryStorage(MongoDatabase mongoDatabase,OptionsContext optionsContext, Document currentDocument,String fieldName,
+			Tuple tuple ) {
+		String gridfsBucketName = optionsContext.getUnique( GridFSBucketOption.class );
+
+		GridFSBucket gridFSFilesBucket = getGridFSFilesBucket( mongoDatabase, gridfsBucketName );
+		String fileName = "";
+		BinaryStream binaryStream = currentDocument.get( fieldName,BinaryStream.class );
+		ObjectId uploadId = gridFSFilesBucket.uploadFromStream( fileName, binaryStream.getInputStream() );
+		//change value of the field (BinaryStream -> ObjectId)
+		currentDocument.put( fieldName, uploadId );
+		if ( Tuple.SnapshotType.UPDATE == tuple.getSnapshotType() ) {
+			ObjectId oldContentObjectId = (ObjectId) tuple.get( fieldName + "_uploadId" );
+			gridFSFilesBucket.delete( oldContentObjectId );
+		}
+	}
+
+	/**
+	 * remove binary content from store
+	 * @param mongoDatabase current instance of MongoDB
+	 * @param optionsContext options context
+	 * @param deletedDocument deleted BSON document
+	 * @param fieldName field name with binary content
+	 */
+
+	@Override
+	public void removeContentFromBinaryStore(MongoDatabase mongoDatabase, OptionsContext optionsContext, Document deletedDocument, String fieldName) {
+		String gridfsBucketName = optionsContext.getUnique( GridFSBucketOption.class );
+		GridFSBucket gridFSFilesBucket = getGridFSFilesBucket( mongoDatabase, gridfsBucketName );
+		ObjectId gridFsLink = deletedDocument.get( fieldName, ObjectId.class );
+		gridFSFilesBucket.delete( gridFsLink );
+	}
+
+	/**
+	 * load content from binary store to field
+	 * @param mongoDatabase current instance of MongoDB
+	 * @param optionsContext options context
+	 * @param currentDocument current BSON document
+	 * @param fieldName field name with binary content
+	 */
+	@Override
+	public void loadContentFromBinaryStorageToField(MongoDatabase mongoDatabase, OptionsContext optionsContext,Document currentDocument,String fieldName) {
+		//the field has GridFS configuration. Process it!
+		String gridfsBucketName = optionsContext.getUnique( GridFSBucketOption.class );
+
+		GridFSBucket gridFSFilesBucket = getGridFSFilesBucket( mongoDatabase, gridfsBucketName );
+
+		ObjectId uploadId = currentDocument.get( fieldName, ObjectId.class );
+		ByteArrayOutputStream fullContent = new ByteArrayOutputStream( 100_000 );
+		//@todo Ask Davide about Blob Proxy
+		//GridFSDownloadStream downloadStream = gridFSFilesBucket.openDownloadStream( uploadId );
+		//read full blob
```

Comment:
Please, don't add this kind of comments in the PR. If you need to ask me about something, ask the question.
Or, at least, remove them before sending a PR because they could be included in the master repository by mistake